bundles/org.openhab.core/src/test/java/org/openhab/core/util/ColorUtilTest.java

```diff
@@ -313,4 +384,241 @@ private void assertRgbEquals(final int[] expected, final int[] actual) {
             assertEquals(expectedS, actualS);
         }
     }
+
+    /**
+     * Test round trips HSB => xyY => HSB
+     */
+    @Test
+    public void hsbToXY2xyToHsb() {
+        boolean assertionExecuted = false;
+        for (double h = 0; h < 360; h = h + 5) {
+            for (double s = 0; s <= 100; s = s + 5) {
+                for (double b = 0; b <= 100; b = b + 5) {
+                    HSBType hsb1 = new HSBType(new DecimalType(h), new PercentType(new BigDecimal(s)),
+                            new PercentType(new BigDecimal(b)));
+                    double[] xyY = new double[3];
+                    HSBType hsb2 = HSBType.BLACK;
+                    try {
+                        xyY = ColorUtil.hsbToXY(hsb1);
+                        hsb2 = ColorUtil.xyToHsb(xyY);
+
+                        // HSB assertions are meaningless if B is zero, or xy was forced into gamut
+                        if (b == 0 || xyY.length > 3) {
+                            continue;
+                        }
+
+                        // assert that S values are within 0.01%
+                        assertEquals(hsb1.getSaturation().doubleValue(), hsb2.getSaturation().doubleValue(), 0.01);
+
+                        // assert that B values are within 0.01%
+                        assertEquals(hsb1.getBrightness().doubleValue(), hsb2.getBrightness().doubleValue(), 0.01);
+
+                        // H assertions are meaningless if S is zero
+                        if (s == 0) {
+                            continue;
+                        }
+
+                        // assert that H values are within 0.05 degrees
+                        double h1 = hsb1.getHue().doubleValue();
+                        h1 = h1 >= 180.0 ? 360.0 - h1 : h1;
+                        double h2 = hsb2.getHue().doubleValue();
+                        h2 = h2 >= 180.0 ? 360.0 - h2 : h2;
+                        assertEquals(h1, h2, 0.05);
+
+                        assertionExecuted = true;
+                    } catch (AssertionError e) {
```

Comment:
Why do you catch it here instead of failing immediately?

Reply:
The purpose is for the error log to display a nicely formatted message with the values that failed. It makes it much easier to figure out which particular arguments version of the case actually caused the error.

```
String.format("HSB1:[%.6f,%.6f,%.6f] - xyY:[%.6f,%.6f,%.6f] - HSB2:[%.6f,%.6f,%.6f] - %s",
```